tagged enums, crates update & latex "rendering"

WhySoBad opened the pull request:
Hi,

I've adjusted the backend to use tagged enums for formulas and rules. This gives better types for the openapi schema and enables a cleaner implementation in the frontend. 

Additionally, I added katex to provide a font for all math blocks (also the unicodes now look as they would in LaTeX)

During experimenting with some things I realized the crates were very outdated. I therefore updated the axum and utoipa related crates.

WhySoBad committed:
feat: refactor and cleanup

WhySoBad committed:
feat: add LaTeX font

WhySoBad committed:
feat: add LaTeX font to matcher

WhySoBad committed:
feat: combined Dockerfile, CI, add logs

The-Ray-Man commented:
Hi
It's great to see that this is an interesting project. Thanks for contributing!
I will review the changes and merge them during the next week :)

WhySoBad committed:
feat: add operator synonyms, add theme switch

WhySoBad commented:
Hi, no stress with the review

Here is a small overview of the changes of the last two commits:
- Created a new docker image which serves both the API and the frontend from the axum web server. Additionally, the image uses alpine and docker build steps which decreases the image size by a lot
- Added synonyms to the operator grammar to allow for latex/typst inputs
- There is now a tooltip in the enter formula input which describes the available operators

Additionally, I've added a light/dark theme switch to the header. However, the highlighted text does not look pretty in the dark theme, maybe the highlight should be changed from the background color to something different (e.g. underline)
